src/hooks/useComponentVisible.tsx

```diff
@@ -0,0 +1,25 @@
+import { useState, useEffect } from 'react';
+
+export default function useComponentVisible(
+	ref: React.RefObject<HTMLElement>,
+	initialIsVisible = false
```

Comment:
Does this work without `const`?

Reply:
I'm not quite sure of what you're asking if you could elaborate a little please!

Reply:
Haha it's just strange seeing what seems to be a variable declaration without `const` or `let`, unless I'm missing something here

Reply:
Ah yes, you are missing one thing. This is the parameter list for `useComponentVisible`!

Ie `function someFunction( initialIsVisible = false )`. This is a way of providing a default value for a function parameter, so you can call the function without specifying a parameter, and it will default to `false`. Really handy!

Read more about it here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters